include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/net/prometheus/pm_metric.h>
+
+#include <stddef.h>
+
+/**
+ * @brief Prometheus collector definition
+ *
+ * This structure defines a Prometheus collector.
+ */
+struct prometheus_collector {
+	/* Name of the collector */
+	const char *name;
+	/* Array of metrics associated with the collector */
+	struct prometheus_metric *metric[CONFIG_PROMETHEUS_MAX_METRICS];
+	/* Number of metrics associated with the collector */
```

Comment:
Please change the comments here to doxygen comments so that the documentation is generated automatically for these. So `/** .... */`